src/extension.ts

```diff
@@ -16,39 +14,33 @@ function ocpIndentPath(): string {
 function ocpCommand() {
 	const path = ocpIndentPath();
 	if (path === ""){
-		return 'ocp-indent --inplace';
+		return 'ocp-indent';
 	} else {
-		return path + ' --inplace';
+		return path;
 	}
 }
 
-function executeOcpIndent(document: vscode.TextDocument, option: string) {
-	if (existsSync(document.uri.fsPath)) {
-		const { path } = document.uri;
-		document.save().then((value) => {
-			const cp = require('child_process');
-			cp.exec(ocpCommand() + ' ' + path + ' ' + option,
-				(err: string, _: string, stderr: string) => {
-					if (stderr) { vscode.window.showErrorMessage(stderr); }
-					if (err) { vscode.window.showErrorMessage(err); }
-				});
-		});
-	} else {
-		vscode.window.showWarningMessage
-			('OCP-Indent can only deal with saved files, save the file and try again.');
-	}
+function executeOcpIndent(document: vscode.TextDocument, args: string[]): string | undefined {
+	const cmd = ocpCommand();
+	const text = document.getText();
+	const options = {
+		input: text,
+		encoding: 'utf8',
```

Comment:
Are we forced to use utf8 here. I mean, what if the file is not encoded in utf8?

Reply:
Unfortunately, I didn't find how to get the encoding of a vscode document, so I don't known how to fix this.